src/slider.jsx

```diff
@@ -428,9 +476,9 @@ const Slider = React.createClass({
 
   _updateWithChangeEvent(e, percent) {
     if (this.state.percent === percent) return;
-    this.setPercent(percent);
-    let value = this._alignValue(this._percentToValue(percent));
-    if (this.props.onChange) this.props.onChange(e, value);
+    this.setPercent(percent, () => {
+      if (this.props.onChange) this.props.onChange(e, this.state.value);
```

Comment:
This will call onChange many times with same value during dragging. I think it will be better to call onChange only when value changes.